### PR TITLE
feat: `cast mktx`

### LINKED ISSUE #1273: cast mktx / sign transactions without broadcasting
### Component

Cast

### Describe the feature you would like

It would be nice if `cast` can sign and prepare transactions without broadcasting them.

This will be especially useful for hardware wallets where a signed transaction is created (via hardware device prompts) and the transaction is broadcasted at a later time

### Additional context

`seth mktx` returns a raw hash which can be used with `seth publish` (and `cast publish`)

https://github.com/dapphub/dapptools/tree/master/src/seth#seth-mktx

### PR DESCRIPTION
**Pull request by ay**
## Motivation

Similar to `seth mktx` from dapptools, there should be a command to build and sign transactions without broadcasting them. Ideally it should be possible to do this on an offline machine.

## Solution

`cast mktx` is the same as `cast send` except it prints out a raw hex encoding of the signed transaction instead of broadcasting. This output can then be published with `cast publish`. It also works without an RPC (for example on an air-gapped machine) if all the required options are set.

I can also update the Foundry book if merged.

Closes #1273

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
I'm supportive, since seth also has this.

I'm fine with the verbose `build_tx` function, because we'll clean this up shortly anyway

could we add a few tests for this

pta closer look @Evalir

File: crates/cast/bin/cmd/mktx.rs
Comment (mds1) [inline crates/cast/bin/cmd/mktx.rs:103]: Since we're actually signing the transaction, this probably should be behind `cast wallet` similar to the existing `cast wallet sign`. What do you think about also renaming this to `cast wallet sign-tx` to make it more explicit? I don't think cast needs to be backwards compatible with seth anymore. 

I could imagine a separate `cast mktx` command that returns a populated (optionally serialized) transaction, without signing it

Comment (ay) [reply crates/cast/bin/cmd/mktx.rs:103]: I went back and forth a few times on this and initially implemented this as `cast wallet sign`, but ultimately decided `cast mktx` is better. The `cast wallet` commands don't use `TxBuilder`, don't use an RPC connection, and therefore have no way of filling a transaction with gas info, nonce, ENS resolution, etc. Having it as `cast mktx` allows it to be as close to `cast send` as possible, with the difference being it prints the encoded transaction instead of broadcasting it. You get everything provided by `TxBuilder` if you need it, but can still specify all the options manually if you have no connection to an RPC endpoint.

Comment (ay) [reply crates/cast/bin/cmd/mktx.rs:103]: I suppose it can be renamed to something like `cast sign-tx`, but since the behavior is identical to `seth mktx` (which builds and signs transactions) I went with `cast mktx`.

Comment (mds1) [reply crates/cast/bin/cmd/mktx.rs:103]: I think a good general principle is to implement the UX that's best for the user, even if it's more work because `cast wallet` commands don't yet have `TxBuilder` or RPC connections. Most foundry users haven't used (or heard of) dapptools and seth is no longer maintained, so we should use the best naming/UX regardless of what seth did.

However, IMO the cast UX is already messy and needs breaking change cleanup, so I won't block this PR on trying to keep cast UX clean or changing command names. Just sharing my thoughts, and will defer to you and other foundry maintainers from here :)

Comment (ay) [reply crates/cast/bin/cmd/mktx.rs:103]: Yeah, I agree. I wasn't sure what the best place for this was either. Because of how close its logic is to `cast send`, I put it adjacent to where that is. I can rename/move it wherever (though moving it under `cast wallet` will require a few more changes like importing `TxBuilder` and the utility functions in common with `cast send`).

**Comment by ay** [comment]
I've added some tests.

**Review by mattsse — APPROVED** [review]
sorry for the delay.

lgtm

the tx builder abomination will we refactored with cast alloy migration, so this is fine

**Pull request merged**
